### ISSUE TITLE
Android Basics: Viewmodel and State in Compose - Write Unit Test codelab

### ISSUE THREAD

**Issue by rexkekeke** [google-developer-training/basic-android-kotlin-compose-training-unscramble#32]
The solution code doesn't pass unit test.

- In `gameViewModel_Initialization_FirstWordLoaded` test, `gameUiState.currentWordCount` should be equal to 0.
- In `updateGameState(updatedScore: Int) ` function in `GameViewModel` class, we missed a `currentWordCount = currentState.currentWordCount.inc(),` at the last round of the game, which cause the `gameViewModel_AllWordsGuessed_UiStateUpdatedCorrectly` test to fail.

**Comment by scolton99** [comment]
Agreed, I thought that I was losing my mind so I'm glad to see I'm not the only one.

**Comment by osuleymanova** [comment]
Duplicate of #31, #30, #19, #12, #4.
